prog/analysis.go

```diff
@@ -309,3 +310,141 @@ func encodeFallbackSignal(typ, id, aux int) uint32 {
 func decodeFallbackSignal(s uint32) (typ, id, aux int) {
 	return int(s & 7), int((s >> 3) & fallbackCallMask), int(s >> 16)
 }
+
+type pair struct {
```

Comment:
This is a too generic name, it can cause conflicts. Give is a more specific name.